src/main/client/src/components/@commons/DataSheet.tsx

```diff
@@ -17,92 +16,118 @@ type DataSheetProps = {
+              </Box>
+            ))}
+          </Flex>
         )}
-      </Wrap>
-    </>
-  )
+      </VStack>
+      {generateLogo(organization)}
+    </Flex>
+    {organization.descriptionParagraphs && generateParagraphs(organization.descriptionParagraphs)}
+
+    <Wrap marginTop={10} justify={{ base: 'center', md: 'flex-end' }}>
+      {organization.website && (
+        <LinkButton href={organization.website} external leftIcon={<LinkIcon />} colorScheme={organization.color}>
+          Weboldal
+        </LinkButton>
+      )}
+      {organization.application && (
+        <LinkButton href={organization.application} external leftIcon={<EditIcon />} colorScheme="brand">
+          Jelentkezés
+        </LinkButton>
+      )}
+      {organization.facebook && (
+        <LinkButton href={organization.facebook} external leftIcon={<FaFacebook />} colorScheme="facebook">
+          Facebook
+        </LinkButton>
+      )}
+      {organization.instagram && (
+        <LinkButton href={organization.instagram} external leftIcon={<FaInstagram />} colorScheme="purple">
+          Instagram
+        </LinkButton>
+      )}
+    </Wrap>
+  </>
+)
+
+type DataFieldProps = {
+  icon: JSX.Element
+  label: string
 }
 
-const DataGrid: React.FC = ({ children }) => {
+const DataField: React.FC<DataFieldProps> = ({ icon, label, children }) => {
   return (
-    <Grid templateColumns="repeat(2,auto)" mt={5} gap={5} width="fit-content" mr={5}>
+    <HStack color={useColorModeValue('gray.700', 'gray.200')}>
+      <Box>{icon}</Box>
+      <Box fontWeight={700}>{label}</Box>
       {children}
-    </Grid>
+    </HStack>
   )
 }
 
-type DataFieldProps = {
-  label: string
+const generateParagraphs = (paragraphs: string | string[]): JSX.Element => {
+  if (Array.isArray(paragraphs)) {
+    return (
+      <>
+        {paragraphs.map((paragraph, index) => (
+          <Paragraph key={`${index}@${paragraph.substring(0, 16)}`}>{paragraph}</Paragraph>
+        ))}
+      </>
+    )
+  }
+
+  return <Paragraph>{paragraphs}</Paragraph>
 }
 
-const DataField: React.FC<DataFieldProps> = ({ label, children }) => {
-  return (
-    <>
-      <GridItem color="gray.500" colStart={1} colEnd={1}>
-        {label}:
-      </GridItem>
-      <GridItem color="gray.500" colStart={2} colEnd={2}>
-        {children}
-      </GridItem>
-    </>
-  )
+const generateLogo = (org: Organization): JSX.Element | null => {
+  let logoSource: string | null | undefined = null
+
+  if (org.logo) {
+    logoSource = org.darkLogo ? useColorModeValue(org.logo, org.darkLogo) : org.logo
+  } else {
+    logoSource = org.darkLogo
+  }
+
+  if (logoSource) {
+    return (
+      <Image
+        my={2}
+        alignSelf={{ base: 'flex-end', sm: 'flex-start' }}
+        src={logoSource}
+        alt={org.name}
+        maxH={{ base: '10rem', sm: '10rem', md: '16rem' }}
+        maxW={{ base: '16rem', sm: '10rem', md: '16rem' }}
+        objectFit="contain"
+      />
+    )
+  }
```

Comment:
Hook-independent code parts can be lifted out into functions to spare the eye strain while reading a React Function Components structure.